README.md

```diff
@@ -1 +1,1148 @@
+a suitable Unix based shell. The exercises include commands which can be run
+in such a shell, and are represented by a line prefixed with `$`. Information
+returned from execution of a command may be appended under the line to show
+what might be returned. For example:
+
+```
+$ balena version
+11.9.3
+```
+
+## 1. Installation and Authentication
+
+First we need to install balenaCLI. Open a terminal on your development machine
+and run the following command:
+```
+$ npm --global install balena-cli
+```
+This will install the balenaCLI globally and allow you to run it in a terminal
+via `balena <command>`.
+
+To use balenaCLI, you need to login to a balenaCloud account (Ensure you have
+signed up for a balenaCloud account
+[here](https://dashboard.balena-cloud.com/signup)). Next, login to balenaCLI
+using your account:
+```
+$ balena login
+```
+
+You will be asked how you wish to authenticate with balenaCloud. The
+easiest method is that of 'Web authorization' which will bring up a browser
+window (and ask you to first login to balenaCloud if you have not) and ask you
+to confirm you wish to login.
+
+Other authentication methods include using your username and password
+credentials or authentication token.
+
+Once logged in, a [JWT](https://jwt.io/) session token will be saved in your
+home directory (`~/.balena/token`). Be aware that the lifetime of a balena
+JWT is limited to seven days, after which time reauthentication will be
+required.
+
+## 2. Creating an Application and Provisioning a Device
+
+### 2.1 Creating an Application
+Applications can be created via the dashboard or via the balenaCLI. We're going
+to create a new application via balenaCLI called `cliApp`. Run the following
+command:
+```
+$ balena app create cliApp
+```
+This will ask you which device type you wish to create the application for.
+You can scroll up and down this list using the arrow keys. For now, exit
+the command by hitting `Ctrl-C`, as there's another, non-interactive way to
+do this which we'll use instead. Type:
+```
+$ balena devices supported
+```
+to see a list of all supported device types by balenaCloud. We'll pass the
+balenaFin device type (`fincm3`) to the application creation command directly:
+```
+$ balena app create cliApp --type fincm3
+Application created: cliApp (fincm3, id 1234567)
+```
+As can be seen, this will return the name of the application, its type
+(`fincm3`) and its unique ID.
+
+Non-interactive commands are useful when you need to script actions via
+balenaCLI for a shellscript (although balena also includes HTTPS endpoints and
+SDKs which can be used for this purpose).
+
+You can list the applications currently owned by (or shared with) your account
+by typing:
+```
+$ balena apps
+ID      APP NAME         DEVICE TYPE      ONLINE DEVICES DEVICE COUNT
+1234567 cliApp           fincm3           0              0
+```
+
+### 2.2 Provisioning a Device
+
+You can now provision your balenaFin by downloading a provisioning image from
```

Comment:
Since this is a CLI masterclass, how about `balena os download` and then hit it with some `balena os configure`?

Reply:
I'm counting that as a bit more of an advanced masterclass, `os download` and `os configure` open a little can of worms as they're not instantly deployable images.